src/IO/S3/Client.cpp

```diff
@@ -574,6 +574,9 @@ Client::doRequest(RequestType & request, RequestFn request_fn) const
         if (!new_uri)
             return result;
 
+        if (initial_endpoint.substr(11) == "amazonaws.com") // Check if user didn't mention any region
```

Comment:
What is 11 symbol prefix?
Maybe it's better to use
https://github.com/ClickHouse/ClickHouse/blob/b10183e7af7ea899ed765a5654ed12dd967fca6f/src/Common/StringUtils/StringUtils.h#L36

Reply:
I think `endsWith()` will not be the best option because we should always have `amazonaws.com` at the end of the endpoint. We should check that we have only the clear endpoint like: `https://s3.amazonaws.com`, but not like this: `https://s3.eu-north-1.amazonaws.com`